packages/beacon-node/src/chain/blocks/verifyBlock.ts

```diff
@@ -38,249 +31,79 @@ export type VerifyBlockModules = {
 
-      case ExecutePayloadStatus.INVALID_BLOCK_HASH:
-      case ExecutePayloadStatus.ELERROR:
-      case ExecutePayloadStatus.UNAVAILABLE:
-        throw new BlockError(block, {
-          code: BlockErrorCode.EXECUTION_ENGINE_ERROR,
-          execStatus: execResult.status,
-          errorMessage: execResult.validationError,
-        });
-    }
+      // All signatures at once
+      verifyBlocksSignatures(chain, preState0, blocks, opts),
 
-    // If this is a merge transition block, check to ensure if it references
-    // a valid terminal PoW block.
-    //
-    // However specs define this check to be run inside forkChoice's onBlock
-    // (https://github.com/ethereum/consensus-specs/blob/dev/specs/bellatrix/fork-choice.md#on_block)
-    // but we perform the check here (as inspired from the lighthouse impl)
-    //
-    // Reasons:
-    //  1. If the block is not valid, we should fail early and not wait till
-    //     forkChoice import.
-    //  2. It makes logical sense to pair it with the block validations and
-    //     deal it with the external services like eth1 tracker here than
-    //     in import block
-    if (isMergeTransitionBlock) {
-      const mergeBlock = block.message as bellatrix.BeaconBlock;
-      const mergeBlockHash = toHexString(
-        chain.config.getForkTypes(mergeBlock.slot).BeaconBlock.hashTreeRoot(mergeBlock)
-      );
-      const powBlockRootHex = toHexString(mergeBlock.body.executionPayload.parentHash);
-      const powBlock = await chain.eth1.getPowBlock(powBlockRootHex).catch((error) => {
-        // Lets just warn the user here, errors if any will be reported on
-        // `assertValidTerminalPowBlock` checks
-        chain.logger.warn(
-          "Error fetching terminal PoW block referred in the merge transition block",
-          {powBlockHash: powBlockRootHex, mergeBlockHash},
-          error
-        );
-        return null;
-      });
-      const powBlockParent =
-        powBlock &&
-        (await chain.eth1.getPowBlock(powBlock.parentHash).catch((error) => {
-          // Lets just warn the user here, errors if any will be reported on
-          // `assertValidTerminalPowBlock` checks
-          chain.logger.warn(
-            "Error fetching parent of the terminal PoW block referred in the merge transition block",
-            {powBlockParentHash: powBlock.parentHash, powBlock: powBlockRootHex, mergeBlockHash},
-            error
-          );
-          return null;
-        }));
+      // Execution payloads
+      verifyBlocksExecutionPayload(chain, blocks, preState0, abortController.signal, opts),
+    ]);
 
-      assertValidTerminalPowBlock(chain.config, mergeBlock, {executionStatus, powBlock, powBlockParent});
+    if (mergeBlockFound !== null) {
+      // merge block found and is fully valid = state transition + signatures + execution payload.
+      // TODO: Will this banner be logged during syncing?
```

Comment:
it'd be nice to log the merge block so remove this `TODO` too?

Reply:
I don't think it should be logged during sync, tho would prefer to debate in a separate issue. This TODO was more to note that this can happen